### ISSUE TITLE
Extend search time statistics

### ISSUE THREAD

**Issue by skweee** [martomi/chiadog#225]
**Background**

At the moment we have calculate the average search time for all searches. This number can be misleading for people running smaller farms. Let me explain:

Search time is relevant to make sure your disk space is fast enough to return proofs in the required time frame of 30 seconds (quicker is better). However, if there are 0 eligible plots for a given challenge, most of the time no disk space access is required, as plot filters will be answered from memory.

The relevant info for disk access time comes from searches involving 1 or more eligble plots. It would therefore make sense to differentiate between searches with 0 eligble plots and searches with 1 or more eligble plots. Only the latter statistics would provide meaningful insights for search times accessing disk space.

Imagine somebody farming only a single plot. Even if they have extremely slow disk access times (let's say 30 seconds), their statistics will look moderately good. Statistically only 1 in every 512 challenges will access their disk. So if searches with 0 eligible plots (RAM searches) will be answered in 0.05 seconds and searches with 1 eligible plots are answered in 25 seconds, the average time will be (511*0.05 + 30) / 512 = 0.108 seconds. Their number of plots over 15s will also look quite low, because only 1 out of 512 searches will access the disk, so only about 0.2 percent of searches will be above 15 seconds.

For a tiny farmer (only 1 plot), the statistics will look great even though with the above numbers they never have a chance to win a block. The more plots this farmer have, the statistics will change an he will see the trouble he is in. Larger farmers (probably >1000 plots) have a high probability of finding eligible plots for all challenges. Thus their average search time will be based almost exclusively on disk access and they will see the trouble they are in. Small farmers however, will see statistics skewed by memory access.

The farmer from this example (only 1 plot) will see the following statistics at the moment:

`Search:
  - average: 0.11s over 9376 searches
  - over 5s: 18 occasions (0.19%)
  - over 15s: 18 occasions (0.19%)`

**Suggestion**

We should differentiate search statistics for 0 eligible plots (mostly RAM access) and 1 or more eligible plots (disk access). For the above example (farmer with only 1 plot), the following statistics would be helpful in diagnosing his issue:

`Search:
  - 9376 searches
  - average (RAM): 0.05s
  - average (disk): 30.00s
  - over 5s: 18 occasions (100.00% of disk searches)
  - over 15s: 18 occasions (100.00% of disk searches)`

**Comment by martomi** [comment]
Good insights! 

I'd suggest to do it a bit simpler and only consider challenges with at least 1 eligible plot for the statistics. The data about 0 eligible plots doesn't really add valuable information as I see it.

**Comment by skweee** [comment]
Good point! That also makes the fix easier and simplifies reading the statistics. Will edit the issue title to reflect this.

**Comment by skweee** [comment]
Looking at the relevant code, I'm not sure whether the statistics are calculated correctly at all at the moment:
https://github.com/martomi/chiadog/blob/429b90cf2d3f74a885d4ffcb748fb63b1521e48b/src/chia_log/handlers/daily_stats/stat_accumulators/search_time_stats.py#L25

Let's assume we process 4 challenges. The first one takes 2 seconds, the next two challenges take 1 second and the last one takes 2 seconds. The average time would be (2+1+1+2)/4 = 1.5 seconds.

If we look at the code, this would happen:

`self._avg_time_seconds += (obj.search_time_seconds - self._avg_time_seconds) / self._num_measurements`

`self._avg_time_seconds += (2-0)/1` --> `result = 2` (this is correct)
`self._avg_time_seconds += (1-2)/2` --> `result = -0.5` (this doesn't make sense)
`self._avg_time_seconds += (1-(-0.5))/3` --> `result = 0.5` (this is still wrong)
`self._avg_time_seconds += (2-0.5)/4` --> `result = 0.375` (this is still wrong)

Either I'm missing something here or we need to find a different approach.

**Comment by skweee** [comment]
Proposing to change this line...
https://github.com/martomi/chiadog/blob/429b90cf2d3f74a885d4ffcb748fb63b1521e48b/src/chia_log/handlers/daily_stats/stat_accumulators/search_time_stats.py#L25

to this:
`self._avg_time_seconds += (obj.search_time_seconds + (self._avg_time_seconds * (self._num_measurements - 1))) / self._num_measurements`

But I'm not sure if I'm missing something, so I'd love to hear some feedback.
